### PR TITLE
feat: Report user UI

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Further comments

[TC-730]

[TC-730]: https://rocketchat.atlassian.net/browse/TC-730?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by guijun13 — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		<Modal wrapperFunction={(props: ComponentProps<typeof Box>) => <Box is='form' onSubmit={handleSubmit(onClickConfirm)} {...props} />}>
+			<Modal.Header>
+				<Modal.Icon name='warning' color='danger' />
+				<Modal.HeaderText fontScale='h3'>{t('Report_User')}</Modal.HeaderText>
+				<Modal.Close onClick={onClose} />
+			</Modal.Header>
+			<Modal.Content>
+				<Box mbe='x16' fontWeight='bold'>
+					<UserAvatar username={username} /> {displayName}
+				</Box>
```
[the remark this answers is not in the document]
Comment (yash-rajpal) [reply apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx]: Done! Thanks

File: apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx
Comment (dougfabris) [inline apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx:1]: what about using `GenericModal`?

File: apps/meteor/client/views/room/hooks/useUserInfoActions/actions/useReportUser.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			onSuccess: () => dispatchToastMessage({ type: 'success', message: t('Report_has_been_sent') }),
+			onError: (error) => dispatchToastMessage({ type: 'error', message: error }),
+			onSettled: () => setModal(),
+		},
+	);
+
+	const openReportUserModal = useMemo(() => {
+		const action = () =>
+			setModal(
+				<ReportUserModal
```
Comment (dougfabris) [inline apps/meteor/client/views/room/hooks/useUserInfoActions/actions/useReportUser.tsx:35]: Let's handle `useDisplayName` here and just pass the return name to the prop to the Modal!

**Review by dougfabris — CHANGES_REQUESTED** [review]
Also, I'm not being able to check the report in the Moderation Console..

File: apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx
Comment (yash-rajpal) [reply apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx:1]: You are right, I missed it. Not sure why I didn't use `GenericModal` 😅

File: apps/meteor/client/views/room/hooks/useUserInfoActions/actions/useReportUser.tsx (hunk 2 above)
Comment (yash-rajpal) [reply apps/meteor/client/views/room/hooks/useUserInfoActions/actions/useReportUser.tsx:37]: Done!

**Comment by yash-rajpal** [comment]
The feature to view reported users is still under development.

**Review by guijun13 — APPROVED** [review]

File: apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		getValues,
+		handleSubmit,
+		formState: { errors },
+	} = useForm<ReportUserModalsFields>({
+		defaultValues: {
+			description: '',
+		},
+	});
+
+	const onClickConfirm = () => {
```
Comment (dougfabris) [inline apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx:33]:
```suggestion
	const onClickConfirm = (description) => {
```

File: apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		const { description } = getValues();
```
Comment (dougfabris) [inline apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx:34]:
```suggestion

```

File: apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+type ReportUserModalsFields = {
+	description: string;
+};
+
+const ReportUserModal = ({ username, displayName, onConfirm, onClose }: ReportUserModalProps) => {
+	const {
+		register,
+		getValues,
```
Comment (dougfabris) [inline apps/meteor/client/views/room/contextualBar/UserInfo/ReportUserModal.tsx:24]:
```suggestion

```

**Review by dougfabris — CHANGES_REQUESTED** [review]

**Review by guijun13 — DISMISSED** [review]

**Review by tassoevan — DISMISSED** [review]

**Review by dougfabris — APPROVED** [review]

**Review by gabriellsh — DISMISSED** [review]
LGTM

**Review by guijun13 — APPROVED** [review]

**Review by tassoevan — APPROVED** [review]

**Pull request merged**